sub/sd_ass.c

```diff
@@ -288,46 +288,53 @@ static int init(struct sd *sd)
     return 0;
 }
 
+static inline bool is_escaped(const char *str, size_t pos)
+{
+    size_t count = 0;
+    while (pos > 0 && str[--pos] == '\\')
+        ++count;
+    return count % 2;
+}
+
 // Check if subtitle has events that would cause it to be animated inside {}
-static bool is_animated(char *s)
+static bool is_animated(const char *str)
 {
-    bool in_tag = false;
-    bool valid_event = false;
-    bool valid_tag = false;
-    while (*s) {
-        if (!in_tag && s[0] == '{')
-            in_tag = true;
-        if (s[0] == '\\') {
-            s++;
-            if (!s[0])
+    const char *begin = str;
+    while ((begin = strchr(begin, '{'))) {
+        if (!begin)
+            return false;
```

Comment:
Unreachable?

Reply:
sure, I was moving code around and this left over.